src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -685,4 +694,18 @@ private EmailWrapper getEmptyEmailAddressedToEmail(String recipient) {
         return email;
     }
 
+    private String generateCoOwnersEmailsLine(String courseId) {
+        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(courseId);
```

Comment:
I believe this will require a datastore retrieval everytime an email requiring this is generated. Can you change the method to take in a list of InstructorAttributes instead (or overload this method) and try to pass in existing list of Instructors whenever possible?